### ISSUE TITLE
DuckDuckGo

### ISSUE THREAD

**Issue by pellaea** [fnando/browser#444]
## Description

I have a user using DuckDuckGo somehow in conjunction with their browser, resulting in their USER_AGENT string containing "DuckDuckGo", which is in bots.yml.

## How to reproduce

require browser
user_agent = "Mozilla/5.0 (Linux; Android 9) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/79.0.3945.136 Mobile Safari/537.36 DuckDuckGo/5"
browser = Browser.new(user_agent, accept_language: "en-us")
puts browser.bot?.inspect

## What do you expect

Should print "false".

## What happened instead

Prints "true".

## Software:

- Rails version: 5.2.3
- Browser gem version: 2.6.1

## Full backtrace

???

## Report

http://user-agent.herokuapp.com/?ua=Mozilla%2F5.0+%28X11%3B+Ubuntu%3B+Linux+x86_64%3B+rv%3A72.0%29+Gecko%2F20100101+Firefox%2F72.0&al=en-US%2Cen%3Bq%3D0.5

**Comment by fnando** [comment]
Added by v4.0.0. ;)

**Comment by justisb** [comment]
This is still an issue as of version 4.2. The user agent provided here comes back as true for `.bot?`, as well as newer versions:

```
2.7.1 :001 > browser = Browser.new('Mozilla/5.0 (iPhone; CPU iPhone OS 13_3_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 DuckDuckGo/7 Safari/605.1.15')
2.7.1 :002 > browser.bot?
 => true 
2.7.1 :003 > browser.duck_duck_go?
 => true 
2.7.1 :004 > browser.bot.name
 => "Duck Duck Go" 
2.7.1 :005 > browser.bot.why?
 => Browser::Bot::KnownBotsMatcher 
```

**Comment by fnando** [comment]
@justisb Huh. This was an oversight of my part. I'm sorry for that. I've fixed it on #477.

**Comment by justisb** [comment]
Awesome, looking forward to the release. I've been running a monkey patch for KnownBotsMatcher that excludes DuckDuckGo.
